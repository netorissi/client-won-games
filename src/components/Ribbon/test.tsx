import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';
import theme from 'styles/theme';

import Ribbon from '.';

describe('<Ribbon/>', () => {
  it('should render the text correctly', () => {
    const { container } = renderWithTheme(<Ribbon>Best Seller</Ribbon>);

    expect(screen.getByText(/Best Seller/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with primary color', () => {
    const { container } = renderWithTheme(<Ribbon>Best Seller</Ribbon>);

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      backgroundColor: theme.colors.primary
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with secondary color', () => {
    const { container } = renderWithTheme(
      <Ribbon color="secondary">Best Seller</Ribbon>
    );

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      backgroundColor: theme.colors.secondary
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with size normal', () => {
    const { container } = renderWithTheme(<Ribbon>Best Seller</Ribbon>);

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      height: '3.6rem',
      fontSize: theme.font.sizes.small
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with size small', () => {
    const { container } = renderWithTheme(
      <Ribbon size="small">Best Seller</Ribbon>
    );

    expect(screen.getByText(/Best Seller/i)).toHaveStyle({
      height: '2.6rem',
      fontSize: theme.font.sizes.xsmall
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
