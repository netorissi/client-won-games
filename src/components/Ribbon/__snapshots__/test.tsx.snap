// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon/> should render the text correctly 1`] = `
.c0 {
  background-color: #F231A5;
  height: 3.6rem;
  font-size: 1.4rem;
}

<div
  class="c0"
  color="primary"
>
  Best Seller
</div>
`;

exports[`<Ribbon/> should render with primary color 1`] = `
.c0 {
  background-color: #F231A5;
  height: 3.6rem;
  font-size: 1.4rem;
}

<div
  class="c0"
  color="primary"
>
  Best Seller
</div>
`;

exports[`<Ribbon/> should render with secondary color 1`] = `
.c0 {
  background-color: #3CD3C1;
  height: 3.6rem;
  font-size: 1.4rem;
}

<div
  class="c0"
  color="secondary"
>
  Best Seller
</div>
`;

exports[`<Ribbon/> should render with size normal 1`] = `
.c0 {
  background-color: #F231A5;
  height: 3.6rem;
  font-size: 1.4rem;
}

<div
  class="c0"
  color="primary"
>
  Best Seller
</div>
`;

exports[`<Ribbon/> should render with size small 1`] = `
.c0 {
  background-color: #F231A5;
  height: 2.6rem;
  font-size: 1.2rem;
}

<div
  class="c0"
  color="primary"
>
  Best Seller
</div>
`;
